bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,537 @@
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.intesis.internal.IntesisConfiguration;
+import org.openhab.binding.intesis.internal.IntesisDynamicStateDescriptionProvider;
+import org.openhab.binding.intesis.internal.IntesisHomeModeEnum;
+import org.openhab.binding.intesis.internal.api.IntesisHomeHttpApi;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Data;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Datapoints;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Descr;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Dp;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Dpval;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Id;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Info;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IntesisHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+@NonNullByDefault
+public class IntesisHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisHomeHandler.class);
+    private final IntesisHomeHttpApi api;
+    private IntesisConfiguration config = new IntesisConfiguration();
+
+    private final Map<String, String> properties = new HashMap<>();
+
+    private IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
```

Comment:
Might as well make this final.
```suggestion
    private final IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
```
Also, please move all final fields above the non-final ones.

Reply:
Done